src/frequenz/sdk/config/_config_managing.py

```diff
@@ -164,10 +164,21 @@ async def _run(self) -> None:
 
         try:
             async for event in file_watcher:
+                if event.type == EventType.DELETE:
+                    if event.path in self._config_paths:
+                        _logger.info(
+                            "%s: The configuration file %s was deleted, ignoring...",
+                            self,
+                            event.path,
+                        )
+                    continue
+
                 # Since we are watching the whole parent directories, we need to make
                 # sure we only react to events related to the configuration files we
                 # are interested in.
-                if not any(event.path.samefile(p) for p in self._config_paths):
```

Comment:
Can't we check here if `event.path` also exists? I think is harder to reason about the code if we split where the types of events are handled.

Reply:
It won't cover the case when event.path was deleted but is in self._config_paths - we should print LOG in that case.  And first if is to cover this case

Reply:
By the way I think we should print `warning` log in this case, not `info` log. I will change it.

Reply:
> I think it would be good to add a test case for this, as it is not trivial to see why DELETE is handled separately,

It should be easy. I will try to write unit test.

Reply:
> Yeah, OK, but it is still tricky, the check for event.path in self._config_paths could be a bit naive, event.path could be textually different from the path  self._config_paths if it is a symlink or something, I think

Oww you are right. I didn't thought about it...

> Maybe it would be safer to keep a map between the user-specified file name and the canonical file name it resolves to.

Ok I could do that, however it won't be `one line` solution, because I think it is ok if not all config files exists on startup.

Reply:
> But it does all this faster then inotify detects change and it is not working :D
> I could add timeout in the meantime, but I worry it might be unstable test and fail sometimes.

Yeah, we are doing something like that in the channels repo:

* https://github.com/frequenz-floss/frequenz-channels-python/blob/b3c98e594310edde687bde4d9b39f4ac370ed676/tests/test_file_watcher_integration.py#L55

No sure if it would be useful to do something similar here. The thing is this is very filesystem/OS dependant, so it would be probably very hard to do tests that cover all the possibilities, but maybe at least testing for the most common case we'll have (linux, polling) should be better than nothing.

Reply:
> Ok I could do that, however it won't be one line solution, because I think it is ok if not all config files exists on startup.

Yeah, I know, and since it changes the whole logic it will required some careful testing. Maybe we should just completely ignore the delete event for now (as in not register to `DELETE` events at all) for now and open a new issue to improve with the learning from this bug/PR in the future. We want to ignore `DELETE` anyways, as it should only be a bug/accident to remove the config file, we will just not be able to print a warning for it.